python/ray/util/sgd/torch/training_operator.py

```diff
@@ -290,7 +305,8 @@ def setup(self, config):
                                  "are registering schedulers. Set this to "
                                  "'manual' if you will be manually stepping "
                                  "the schedulers.")
-            if len(self._schedulers) == 1:
+            if len(self._schedulers) == 1 and not isinstance(
```

Comment:
what's the purpose of this iterable check? is there a way to abstract this check into some helper functions to improve readability?

Reply:
The reason for the iterable check is to return the output in the same format that the input is passed in. For example, if the user passes in an optimizer object then we return an optimizer object. If the user passes in a list of just a single optimizer, we should return a list of a single optimizer, even though theres just one.

Reply:
it'd be great to just have a:

if check_single_item(self._schedulers, schedulers):

and use it across these different checks

Reply:
yeah makes sense. i will add that.

Reply:
added helper method.